GitCommands/Git/Tag/GitTagController.cs

```diff
@@ -42,7 +51,17 @@ public GitCreateTagCmd GetCreateTagCommand(GitCreateTagArgs args)
             }
 
             var createTagCmd = new GitCreateTagCmd(args, tagMessageFileName);
-            return createTagCmd;
+            try
+            {
+                return presentUI(createTagCmd);
+            }
+            finally
+            {
+                if(tagMessageFileName != null && _fileSystem.File.Exists(tagMessageFileName))
+                {
+                    _fileSystem.File.Delete(tagMessageFileName);
+                }
+            }
```

Comment:
It is a good start, but I think we can do better than this.
I don't think UI must be concerned with how a tag is created, all it must be concerned with - the tag is created or not.
So your initial proposal is in the right direction - abstract tag file handling in the controller, however I don't like the proposed API.

Can we perhaps do something like the following?

```
    public class GitTagController : IGitTagController
    {
        private readonly IGitUICommands _uiCommand;

        public GitTagController(Func<string> getWorkingDir, IGitUICommands uiCommand, IFileSystem fileSystem)
        {
            _uiCommand = uiCommand;
        }

        public bool CreateTag(Form owner, GitCreateTagArgs args)
        {
            // create tag operations

            var success = _uiCommand.StartCommandLineProcessDialog(createTagCmd, owner);

            // if success - delete the tag file

            return success;
        }
    }
```

Reply:
Yes I agree that the ui shouldn't need to have knowledge of that command object. I'll make those changes. Just to be sure though the temp file needs to be deleted regardless of success right?

Reply:
Yes, I suppose that is correct

Reply:
The method must be renamed as well to reflect its new purpose

Reply:
Doh. Ha that seems obvious!